### PR TITLE
Adds CLI for Faros CE

### PR DESCRIPTION
**Pull request by ypc-faros**
# Description

Adds a CLI to quickly set up and sync a few vcs sources.

The CLI can be invoked with no parameters and it will show a number of prompts for the user to choose a source and then source specific parameters (e.g., for github, a github token and repos to sync).

The CLI can also be invoked with subcommands (one per source: github, bitbucket (+ gitlab, jira coming soon) ). One can enter some of the source specific arguments (e.g., `--token` for github) and the corresponding prompt for that parameter will be skipped.

The integration tests rely on a per-source json configuration to be specified as secrets in the github workflows. Those can also be run manually:

`GITHUB_TOKEN=XYZ  npm run test:integration -- --coverage`

Changes to follow in upcoming PRs:
- Support for jira and gitlab.

Fixes # (issue)

## Type of change
(Delete what does not apply)
- New feature (non-breaking change which adds functionality)
- This change requires a documentation update

# Checklist
(Delete what does not apply)
- [X] Have you checked to there aren't other open Pull Requests for the same update/change?
- [X] Have you lint your code locally before submission?
- [X] Have you added an explanation of what your changes do and why you'd like us to include them?
- [X] Have you written new tests for your core changes, as applicable?
- [X] Have you successfully run tests with your changes locally?

### CODE REVIEW

File: cli/test/integration-tests/types.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+export interface TestDefinition {
```
Comment (ypc-faros) [inline cli/test/integration-tests/types.ts:1]: Remove file

File: demo.sh [hunk 2]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+
+git clone https://github.com/faros-ai/faros-community-edition.git
```
Comment (thomas-gerber) [inline demo.sh:3]: improvement suggestion (could be a separate PR): instead of clone the repo, one could just curl the compose, env, metabase folder, start script. I think the rest is not needed (already packaged in containers that will be pulled).

Comment (thomas-gerber) [reply demo.sh:3]: Also, we should call this `install.sh`

Comment (tovbinm) [reply demo.sh:3]: or `setup.sh`

File: .github/workflows/pr.yml [hunk 3]
```diff
@@ -41,10 +41,14 @@ jobs:
             echo "No drift detected"
           fi
 
-      - name: Build
+      - name: Build (init scripts)
```
Comment (tovbinm) [inline .github/workflows/pr.yml:44]: `Build (init)`

File: .github/workflows/pr.yml [hunk 4]
```diff
@@ -70,10 +74,16 @@ jobs:
           echo "DESTINATION_ID=$(cat ${{ github.workspace }}/init/resources/airbyte/workspace/airbyte_config/DESTINATION_CONNECTION.yaml | yq '.[0].destinationId')" >> "$GITHUB_ENV"
           echo $(cat .env | grep "^HASURA_GRAPHQL_ADMIN_SECRET") >> "$GITHUB_ENV"
 
-      - name: Run integration tests
+      - name: Run integration tests (init scripts)
```
Comment (tovbinm) [inline .github/workflows/pr.yml:77]: `Run integration tests (init)`

File: .github/workflows/pr.yml [hunk 5]
```diff
@@ -41,10 +41,14 @@ jobs:
             echo "No drift detected"
           fi
 
-      - name: Build
+      - name: Build (init scripts)
         run: npm ci --unsafe-perm
         working-directory: init
 
+      - name: Build (cli)
```
Comment (tovbinm) [inline .github/workflows/pr.yml:48]: perhaps it's worth splitting this into two parallel jobs now - one for init and another for cli

Comment (tovbinm) [reply .github/workflows/pr.yml:48]: of you use the CE is both? then it might make sense to keep them together, since it's quite time consuming to start the CE in each one of them

Comment (ypc-faros) [reply .github/workflows/pr.yml:48]: Yeah exactly, that's why I kept them together.

File: demo.sh (hunk 2 above)
Comment (ypc-faros) [reply demo.sh]: > improvement suggestion (could be a separate PR): instead of clone the repo, one could just curl the compose, env, metabase folder, start script. I think the rest is not needed (already packaged in containers that will be pulled).

Sounds good! I'll give this a try in another PR

**Review by thomas-gerber — APPROVED** [review]
OK to merge
We will add in separate PRs:
1. better explanation of how the space bar needs to be used to select
2. limits on repos/projects
3. Jira / Bitbucket support
4. replace clone with some curl downloads
